transmogrifier/sources/marc.py

```diff
@@ -0,0 +1,225 @@
+                            value=alternate_title_value.rstrip(".,/"),
+                            kind=alternate_title_marc_field["kind"],
+                        )
+                    )
+
+        # call_numbers
+        call_number_marc_fields = [
+            {
+                "tag": "050",
+                "subfields": "a",
+            },
+            {
+                "tag": "082",
+                "subfields": "a",
+            },
+        ]
+        for call_number_marc_field in call_number_marc_fields:
+            for datafield in xml.find_all(
+                "datafield", tag=call_number_marc_field["tag"]
+            ):
+                for call_number_value in self.create_subfield_value_list_from_datafield(
+                    datafield,
+                    call_number_marc_field["subfields"],
+                ):
+                    fields.setdefault("call_numbers", []).append(call_number_value)
+
+        # citation
+
+        # content_type
+
+        # contents
+
+        # contributors
+
+        # dates
+
+        # edition
+
+        # file_formats
+
+        # format
+
+        # funding_information
+
+        # holdings
+
+        # identifiers
+
+        # languages
+
+        # links
+
+        # literary_form
+        # Literary form is applicable to Book (BK) material configurations and indicated
+        # by leader "Type of Record" position = "Language Material" or "Manuscript
+        # language material" and "Bibliographic level" position =
+        # "Monographic component part," "Collection," "Subunit," or "Monograph/Item."
+        if leader := xml.find("leader", string=True):
+            if leader.string[6:7] in "at" and leader.string[7:8] in "acdm":
+                if fixed_length_data := xml.find(
+                    "controlfield", tag="008", string=True
+                ):
+                    if fixed_length_data.string[33:34] in "0se":
+                        fields["literary_form"] = "Nonfiction"
+                    elif fixed_length_data.string[33:34]:
+                        fields["literary_form"] = "Fiction"
+
+        # locations
+
+        # notes
+
+        # numbering
+
+        # physical_description
+
+        # publication_frequency
+
+        # publication_information
+
+        # related_items
+
+        # rights
+
+        # subjects
+
+        # summary
+
+        return fields
+
+    @staticmethod
+    def get_main_titles(xml: Tag) -> list[str]:
+        """
+        Retrieve main title(s) from a MARC XML record.
+
+        Overrides metaclass get_main_titles() method.
+
+        Args:
+            xml: A BeautifulSoup Tag representing a single MARC XML record.
+        """
+        try:
+            main_title_values = []
+            if main_title_value := Marc.create_subfield_value_string_from_datafield(
+                xml.find("datafield", tag="245"), "abfgknps", " "
+            ):
+                main_title_values.append(main_title_value.rstrip(".,/"))
```

Comment:
I think we also need to include spaces here as many records now have trailing whitespace in the title output. Probably true for alternate titles too.

Reply:
Sorry, mistakenly thought whitespace was included by default, fixed!